packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,398 @@
+
+```tsx
+//usage:
+<DialogFooter
+  primaryAction={<Button>Confirm</Button>}
+  secondaryAction={<Button>Cancel</Button>}
+  tertiaryAction={<Button>Learn more</Button>}
+/>
+```
+
+```html
+<!-- DOM -->
+<div class="fui-dialog-footer">
+  <button>Learn more</button>
+  <button>Confirm</button>
+  <button>Cancel</button>
+</div>
+```
+
+### API
+
+| Property          | Values | Default     | Purpose                                                       |
+| ----------------- | ------ | ----------- | ------------------------------------------------------------- |
+| `primaryAction`   | _slot_ | `undefined` | Primary action that is typically confirmation action.         |
+| `secondaryAction` | _slot_ | `undefined` | Secondary action that is typically the cancel/dismiss action. |
+| `tertiaryAction`  | _slot_ | `undefined` | Tertiary action                                               |
+
+> ⚠️ _Pending issue: should there be any handling for the order of buttons? This also, includes the focus sequence of the buttons when the dialog has a tertiary button as well._
+
+## Dialog Content
+
+The content is a simple container where the content of the dialog is rendered. This component should be used when a trigger is present.
+
+```html
+<div class="fui-dialog-content">{children}</div>
+```
+
+## Dialog Trigger
+
+The trigger is a utility component which is used to control the open/dismiss of the dialog. The main purpose of the trigger is to provide the correct aria values to the trigger button, focus restoration when dialog is closed and offer a way for dialog to be an uncontrolled component. This component will render a button which will control the open/dismiss of the dialog.
+
+### DOM
+
+```html
+<button aria-haspopup="dialog">Open Dialog</button>
+```
+
+### API
+
+```ts
+export type DialogTriggerProps = {
+  /**
+   * Explicitly require single child
+   */
+  children: React.ReactElement;
+};
+```
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+### App
+
+```tsx
+// Custom trigger
+<Button onClick={() => setIsOpen(true)} aria-haspopup="dialog">
+  Open Dialog
+</Button>
 
-## Variants
+<Dialog open={isOpen}>
+  <DialogHeader onClose={() => alert("dialog closed")}>Missing Subject</DialogHeader>
+  <DialogBody>Do you want to send this message without a subject?</DialogBody>
+  <DialogFooter
+    primaryAction={<Button>Submit</Button>}
+    secondaryAction={<Button>Cancel</Button>} />
+</Dialog>
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+// Using the dialog trigger
+<Dialog>
+  <DialogTrigger> Open Dialog </DialogTrigger>
 
-## API
+  {open => (
+    <DialogContent>
+      <DialogHeader>Missing Subject</DialogHeader>
+      <DialogBody>Do you want to send this message without a subject?</DialogBody>
+      <DialogFooter primaryAction={<Button onClick={open}>Submit</Button>} />
+    </DialogContent>
+  )}
```

Comment:
`<Button onClick={open} />`

what is that ^^ ?

I'm not against using a render function for the `DialogContent` child. However, I think it should still support the below example to be consistent

```tsx
<Dialog>
  <DialogTrigger><button /></Dialog>
  <DialogContent />
</Dialog>
```